vault/identity_store_schema.go

```diff
@@ -0,0 +1,117 @@
+package vault
+
+import (
+	"fmt"
+
+	memdb "github.com/hashicorp/go-memdb"
+)
+
+func identityStoreSchema() *memdb.DBSchema {
+	iStoreSchema := &memdb.DBSchema{
+		Tables: make(map[string]*memdb.TableSchema),
+	}
+
+	schemas := []func() *memdb.TableSchema{
+		identityTableSchema,
+		entityTableSchema,
+	}
+
+	for _, schemaFunc := range schemas {
+		schema := schemaFunc()
+		if _, ok := iStoreSchema.Tables[schema.Name]; ok {
+			panic(fmt.Sprintf("duplicate table name: %s", schema.Name))
+		}
+		iStoreSchema.Tables[schema.Name] = schema
+	}
+
+	return iStoreSchema
+}
+
+func identityTableSchema() *memdb.TableSchema {
+	return &memdb.TableSchema{
+		Name: "identities",
+		Indexes: map[string]*memdb.IndexSchema{
+			"id": &memdb.IndexSchema{
+				Name:   "id",
+				Unique: true,
+				Indexer: &memdb.StringFieldIndex{
+					Field: "ID",
+				},
+			},
+			"entity_id": &memdb.IndexSchema{
+				Name:   "entity_id",
+				Unique: false,
+				Indexer: &memdb.StringFieldIndex{
+					Field: "EntityID",
+				},
+			},
+			"mount_type": &memdb.IndexSchema{
```

Comment:
I am inclined towards a `no`. Because, mount ID is unique regardless of the mount type and it is both a necessary and sufficient information to address the use cases which factors address. The use case are 1) Enable same identity names between different mounts 2) Distinguish specific mounts despite mount-unmount-mount on the same path name. MountType plays no role in both of these intentions.